src/watcher/chokidar.ts

```diff
@@ -19,14 +19,18 @@ export type FileWatcher = chokidar.FSWatcher & {
 export type FileWatcherEventCallback = (
   eventName: 'add' | 'addDir' | 'change' | 'unlink' | 'unlinkDir',
   path: string,
-  stats?: fs.Stats
+  stats: fs.Stats | undefined,
+  runner: {
+    restart: (file: string) => void /* stop: () => void, start: () => void */
+  } //TODO: add stop and start methods
 ) => void
 
-type FileWatcherOptions = chokidar.WatchOptions & {
-  onAll?: FileWatcherEventCallback
-}
+type FileWatcherOptions = chokidar.WatchOptions
+const SILENT_EVENTS = ['add', 'addDir'] as const
```

Comment:
More of a comment on how we did things to begin with. Feels now that it would be clearer to call this `SILENTABLE_EVENTS`? And so later: `isSilentableEvent` etc.

Reply:
Totally agreed, fixed 👍